packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -218,6 +197,55 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
     }
 }
 
+/**
+ * TaggedLoggerAdapter class can add tag handling to your logger.
+ */
+ export class TaggedLoggerAdapter implements ITelemetryBaseLogger {
+    public constructor(
+        private readonly logger: ITelemetryBaseLogger) {
+    }
+
+    public send(eventWithTagsMaybe: ITelemetryBaseEvent) {
+        const newEvent: ITelemetryBaseEvent = {
+            category: eventWithTagsMaybe.category,
+            eventName: eventWithTagsMaybe.eventName,
+        };
+        for (const key of Object.keys(getValidTelemetryProps(eventWithTagsMaybe, new Set()))) {
```

Comment:
After thinking it over, I don't think we should call `getValidTelemetryProps` here.  It's overly paranoid - it would only kick in if someone used bad casting to shove an object onto an `ITelemetryProperties`.  And if they did and we _don't_ include this call, it'll still work out - `value` and `tag` will probably be `undefined` which is harmless, or even if they're set to something else, it's a bug and we don't need to guard against it.

That function (getValidTelemetryProps) was written for `LoggingError` where it's easy to add something to an error object needed at runtime, not thinking about the fact that it will also try to be logged (e.g. this happened with ODSP error facet codes). But here we're just copying from event to event and the type system should be good enough.

Reply:
The other thought experiment that convinces me is:  Should we be calling `getValidTelemetryProps` in `TelemetryLogger.prepareEvent` or some similar place?  No, we will trust the type system and people to not go out of their way to shoot themselves in the foot.

Reply:
Makes sense, not an adversarial scenario